managed/ui/src/components/tables/RestoreBackup/RestoreBackup.js

```diff
@@ -116,19 +125,42 @@ export default class RestoreBackup extends Component {
       const labelName = config.metadata.provider + ' - ' + config.metadata.name;
       return { value: config.metadata.configUUID, label: labelName };
     });
+    
+    let configTypeList = <option />;
+    const optGroups =
+      storageConfigs
+      && isNonEmptyArray(storageConfigs)
+      && storageConfigs.reduce((val, indx) => {
+        const configType = `${indx.name} Storage`;
+        val[configType]
+          ? val[configType].push({
+            "configName": indx.configName,
+            "configUUID": indx.configUUID
+          })
+          : (val[configType] = [{
+            "configName": indx.configName,
+            "configUUID": indx.configUUID
+          }]);
+        return val;
+    }, {});
 
-    const storageOptions = storageConfigs.map((config) => {
-      return { value: config.configUUID, label: config.name + ' Storage' };
-    });
+    if (isNonEmptyObject(optGroups)) {
+      configTypeList = Object.keys(optGroups).map(function (key, idx) {
+        return (
+          <optgroup label={key.toUpperCase()} key={key + idx}>
+            {optGroups[key]
+              .sort((a, b) => /\d+(?!\.)/.exec(a) - /\d+(?!\.)/.exec(b))
```

Comment:
code looks very similar to one from `CreateBackup.js` - can this be optimized?

Reply:
Done